consumer/consumererror/signalerrors.go

```diff
@@ -24,20 +33,44 @@ func (err retryable[V]) Data() V {
 	return err.data
 }
 
+// Delay returns the time duration before the telemetry data should
+// be resent to the destination.
+func (err retryable[V]) Delay() time.Duration {
+	return err.delay
+}
+
+// RetryOption allows adding data to a retryable error,
+// e.g. the duration before sending should be retried.
+type RetryOption func(err *retryableCommon)
+
+func WithRetryDelay(delay time.Duration) RetryOption {
+	return func(err *retryableCommon) {
+		err.delay = delay
+	}
+}
+
 // Traces is an error that may carry associated Trace data for a subset of received data
 // that failed to be processed or sent.
 type Traces struct {
 	retryable[ptrace.Traces]
 }
 
 // NewTraces creates a Traces that can encapsulate received data that failed to be processed or sent.
-func NewTraces(err error, data ptrace.Traces) error {
-	return Traces{
+func NewTraces(err error, data ptrace.Traces, options ...RetryOption) error {
```

Comment:
Maybe it's a good opportunity to rename it to `NewRetryable[V ptrace.Traces | pmetric.Metrics | plog.Logs]` and remove `Traces|Metrics|Logs` in favor of generic `Retryable[V ptrace.Traces | pmetric.Metrics | plog.Logs]` type? 

@bogdandrutu WDYT?

With all the other error types we introduce in this PR, I find the assumption that `consumer error.Traces` means a retriable error not that apparent anymore.

Reply:
@dmitryax I think we'd need to have a larger discussion about introducing generics into the public API. I don't think we do it anyone in Core already, but OTTL does. This renames feels like something we could discuss as a followup issue, and wouldn't need to change as part of this PR.

Reply:
> I think we'd need to have a larger discussion about introducing generics into the public API. I don't think we do it anyone in Core already, but OTTL does.

I don't think we should restrict ourselves from introducing generics. It's been two years since they were introduced in Go.

> This renames feels like something we could discuss as a followup issue, and wouldn't need to change as part of this PR.

Ok, sounds good to me. Adding `RetryOption` is not that impactful to bring another breaking change with it.